vault/resource_secrets_sync_association.go

```diff
@@ -0,0 +1,182 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package vault
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/hashicorp/terraform-provider-vault/internal/consts"
+	"github.com/hashicorp/terraform-provider-vault/internal/provider"
+)
+
+const (
+	fieldSecretName = "secret_name"
+	fieldSyncStatus = "sync_status"
+	fieldAccessor   = "accessor"
+	fieldUpdatedAt  = "updated_at"
+)
+
+var secretsSyncAssociationFields = []string{
+	fieldSecretName,
+	fieldSyncStatus,
+	fieldAccessor,
+	fieldUpdatedAt,
+}
+
+func secretsSyncAssociationResource() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: provider.MountCreateContextWrapper(secretsSyncAssociationWrite, provider.VaultVersion115),
+		ReadContext:   provider.ReadContextWrapper(secretsSyncAssociationRead),
+		DeleteContext: secretsSyncAssociationDelete,
+
+		Schema: map[string]*schema.Schema{
+			consts.FieldName: {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Name of the store/destination.",
+				ForceNew:    true,
+			},
+			consts.FieldType: {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Type of secrets store.",
+				ForceNew:    true,
+			},
+			consts.FieldMount: {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Specifies the mount where the secret is located.",
+			},
+			fieldAccessor: {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				Description:  "Specifies the unique accessor of the mount.",
+				ValidateFunc: provider.ValidateNoLeadingTrailingSlashes,
+			},
+			fieldSecretName: {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Specifies the name of the secret to synchronize.",
+			},
+			fieldSyncStatus: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Specifies the status of the association.",
+			},
+			fieldUpdatedAt: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Duration string stating when the secret was last updated.",
+			},
```

Comment:
Out of curiosity are these necessary? Checking my local statefile I can see they are NULL at the moment so it doesn't seem like we persist the computed value.

Reply:
Fixed a bug where these were not being set. These should now be set to the TF state appropriately. I thought them as nice-to-have computed values giving the user more info about their secret if they so desire